dev-tools/omdb/src/bin/omdb/db.rs

```diff
@@ -2249,6 +2416,1221 @@ async fn cmd_db_network_list_vnics(
+    // will print spaces in order to make each column the same size
+
+    let row_char_counts: Vec<_> = rows
+        .iter()
+        .map(|x| {
+            (
+                format!("{}", x.saga_id).chars().count(),
+                x.event_time.chars().count(),
+                if let Some(sub_saga_id) = x.sub_saga_id {
+                    format!("{}", sub_saga_id).chars().count()
+                } else {
+                    0
+                },
+                x.node_id.chars().count(),
+                x.event_type.chars().count(),
+                x.data.chars().count(),
+            )
+        })
+        .collect();
+
+    let (width0, width1, width2, width3, width4): (
+        usize,
+        usize,
+        usize,
+        usize,
+        usize,
+    ) = (
+        row_char_counts.iter().map(|x| x.0).max().unwrap(),
+        row_char_counts.iter().map(|x| x.1).max().unwrap(),
+        std::cmp::max(
+            row_char_counts.iter().map(|x| x.2).max().unwrap(),
+            "sub saga".len(),
+        ),
+        std::cmp::max(
+            row_char_counts.iter().map(|x| x.3).max().unwrap(),
+            "node id".len(),
+        ),
+        std::cmp::max(
+            row_char_counts.iter().map(|x| x.4).max().unwrap(),
+            "event type".len(),
+        ),
+    );
+
+    println!(
+        "{:>width0$} | {:width1$} | {:width2$} | {:width3$} | {:width4$} | {}",
+        String::from("saga id"),
+        String::from("event time"),
+        String::from("sub saga"),
+        String::from("node id"),
+        String::from("event type"),
+        String::from("data"),
+    );
+
+    println!(
+        "{:>width0$} | {:width1$} | {:width2$} | {:width3$} | {:width4$} | {}",
+        (0..width0).map(|_| "-").collect::<String>(),
+        (0..width1).map(|_| "-").collect::<String>(),
+        (0..width2).map(|_| "-").collect::<String>(),
+        (0..width3).map(|_| "-").collect::<String>(),
+        (0..width4).map(|_| "-").collect::<String>(),
+        String::from("---"),
+    );
+
+    for row in rows {
+        println!(
+            "{:>width0$} | {:width1$} | {:>width2$} | {:width3$} | {:width4$} | {}",
+            row.saga_id,
+            row.event_time,
+            if let Some(sub_saga_id) = row.sub_saga_id {
+                format!("{}", sub_saga_id)
+            } else {
+                String::from("")
+            },
+            row.node_id,
+            row.event_type,
+            row.data,
+        );
+    }
+}
+
+/// List all sagas
+async fn cmd_db_sagas_list(
+    datastore: &DataStore,
+    fetch_opts: &DbFetchOptions,
+    args: &SagaListArgs,
+) -> Result<(), anyhow::Error> {
+    let sagas = datastore
+        .pool_connection_for_tests()
+        .await?
+        .transaction_async(|conn| async move {
+            use db::schema::saga::dsl;
+
+            // Sorting by time_created requires a full table scan with the
+            // current index definitions.
```

Comment:
I think I'd just add the index here.  It would really suck if we ran this on a production system that had run a lot of sagas and it just ate up a bunch of database memory or tied up resources for a while executing the query.